src/main/java/edu/cornell/kfs/module/purap/rest/resource/CuEinvoiceApiResource.java

```diff
@@ -233,6 +253,29 @@ private Properties convertUnitOfMeasureToProperties(UnitOfMeasure unitOfMeasure)
         return unitProps;
     }
 
+    private String serializePoShippingAddressToJson(PurchaseOrderDocument poDoc) {
+        Properties addressProps = new Properties();
+        safelyAddProperty(addressProps, CUPurapConstants.Einvoice.DOCUMENT_NUMBER, poDoc.getDocumentNumber());
+        safelyAddProperty(addressProps, CUPurapConstants.Einvoice.SHIPPING_NAME, poDoc.getDeliveryToName());
+        if (StringUtils.isNotBlank(poDoc.getDeliveryBuildingCode())) {
+            safelyAddProperty(addressProps, CUPurapConstants.Einvoice.SHIPPING_DELIVER_TO, poDoc.getDeliveryBuildingCode());
+        }
+        safelyAddProperty(addressProps, CUPurapConstants.Einvoice.ADDRESS_LINE1, poDoc.getDeliveryBuildingLine1Address());
+        safelyAddProperty(addressProps, CUPurapConstants.Einvoice.ADDRESS_LINE2, poDoc.getDeliveryBuildingLine2Address());
+        safelyAddProperty(addressProps, CUPurapConstants.Einvoice.ROOM_NUMBER_LINE, buildRoomNumberLine(poDoc));
+        safelyAddProperty(addressProps, CUPurapConstants.Einvoice.CITY, poDoc.getDeliveryCityName());
+        safelyAddProperty(addressProps, CUPurapConstants.Einvoice.STATE, poDoc.getDeliveryStateCode());
+        safelyAddProperty(addressProps, CUPurapConstants.Einvoice.ZIPCODE, poDoc.getDeliveryPostalCode());
+        safelyAddProperty(addressProps, CUPurapConstants.Einvoice.COUNTRY_CODE, poDoc.getDeliveryCountryCode());
+        safelyAddProperty(addressProps, CUPurapConstants.Einvoice.COUNTRY_NAME, poDoc.getDeliveryCountryName());
+        safelyAddProperty(addressProps, CUPurapConstants.Einvoice.EMAIL, poDoc.getDeliveryToEmailAddress());
+        return gson.toJson(addressProps);
+    }
+
+    private String buildRoomNumberLine(PurchaseOrderDocument poDoc) {
+        return CUPurapConstants.Einvoice.ROOM_NUMBER_PREFIX + poDoc.getDeliveryBuildingRoomNumber();
+    }
```

Comment:
This function adds 3-4 lines of code, but makes the code that calls it a little bit cleaner. I am fine with it either way.

Reply:
I would prefer to keep it this way for now, since it simplifies the calling code a bit.